speechbrain/pretrained/interfaces.py

```diff
@@ -705,8 +705,16 @@ def separate_file(self, path, savedir="."):
         """
         source, fl = split_path(path)
         path = fetch(fl, source=source, savedir=savedir)
+        batch, fs = torchaudio.load(path)
+
+        # resample the data if needed
+        data_folder = self.hparams.data_folder
+        if ("wsj" in data_folder or "wham" in data_folder) and fs != 8000:
```

Comment:
This looks weird to be. I think we should only check if the actual sampling frequency is the one specified in the yaml file, right? This class should be dataset-independent...